pageserver/src/layered_repository/README.md

```diff
@@ -54,8 +75,52 @@ TODO: Are there any exceptions to this?
 For example, timeline.list_rels(lsn) will return all segments that are visible in this timeline at the LSN,
 including ones that were not modified in this timeline and thus don't have a layer in the timeline's LayerMap.
 
-TODO:
-Describe GC and checkpoint interval settings.
+
+# Different kinds of layers
+
+A layer can be in different states:
+
+- Open - a layer where new WAL records can be appended to.
+- Closed - a layer that is read-only, no new WAL records can be appended to it
+- Historical: synonym for closed
+- InMemory: A layer that is kept only in memory, and needs to be rebuilt from WAL
+  on pageserver start
+- OnDisk: A layer that is stored on disk. If its end-LSN is older than
+  disk_consistent_lsn, it is known to be fully flushed and fsync'd to local disk.
```

Comment:
Do we ever use OnDisk layers that are not fully flushed?

Reply:
No, we wait for it to be fsync'd. https://github.com/zenithdb/zenith/pull/815/ will change that, though. After that, we will swap the layers in the layer map as soon as the on-disk layer has been written out to the file, but we don't immediately fsync it. That's done only later.